src/http_server.cpp

```diff
@@ -301,20 +305,28 @@ namespace sim
 
 		http_request req = parse_request(m_recv_buffer.data(), req_len);
 
+		m_recv_buffer.erase(m_recv_buffer.begin(), m_recv_buffer.begin() + req_len);
+		m_bytes_used -= req_len;
+
 		auto it = m_handlers.find(req.path);
 		if (it == m_handlers.end())
 		{
-			// no handler found, 404
-			m_send_buffer = send_response(404, "Not Found");
+			if (std::find(m_stall_handlers.begin(), 
```

Comment:
in which case this becomes: ``if (m_stall_handlers.count(req.path)) return;``
no need to indent the the 404 response or wrap it in an else-block